### PR TITLE
MINOR: Add information to upgrade notes

### PR DESCRIPTION
**Pull request by ijuma**
Credit to Gwen for some of the text.

### CODE REVIEW

**Comment by ijuma** [comment]
Review by @gwenshap

File: docs/upgrade.html [hunk 1]
```diff
@@ -31,13 +31,15 @@
...
 
 <ul>
     <li> Java 1.6 is no longer supported. </li>
     <li> Scala 2.9 is no longer supported. </li>
+    <li> Broker IDs above 1000 are now reserved by default to automatically assigned broker IDs. If your cluster has existing broker IDs above that threshold make sure to increase the reserved.broker.max.id broker configuration property accordingly. </li>
+    <li> Configuration parameter replica.lag.max.messages was removed. Partition leaders will no longer consider the number of lagging messages when deciding which replicas are in sync. </li>
+    <li> Configuration parameter replica.lag.time.max.ms now refers not just to the time passed since last fetch request from replica, but also to time since the replica last caught up. Replicas that are still fetching messages from leaders but did not catch up to the latest messages in replica.lag.time.max.ms will be considered out of sync. </li>
+    <li> MirroMaker no longer supports multiple target clusters. As a result it will only accept a single --consumer.config parameter. To mirror multiple source clusters, you will need at least one MirrorMaker instance per source cluster, each with its own consumer configuration. </li>
```
Comment (guozhangwang) [inline docs/upgrade.html]: @miguno pointed there is a typo here: "MirrorMaker"

Comment (ijuma) [inline docs/upgrade.html]: Thanks @guozhangwang, fixed.

**Comment by guozhangwang** [comment]
LGTM.

**Comment by ewencp** [comment]
LGTM too; merging since it looks like @guozhangwang has not yet.

**Pull request closed without merging**